src/main/java/app/AppBuilder.java

```diff
@@ -2,148 +2,211 @@
-        ServiceManager.registerService(InMemoryStockDataAccessObject.class, stockDataAccessObject);
-
-        // Step 1: Initialize and register InMemoryUserDataAccessObject (DAO)
-        InMemoryUserDataAccessObject userDataAccessObject = new InMemoryUserDataAccessObject();
-        ServiceManager.registerService(InMemoryUserDataAccessObject.class, userDataAccessObject);
-
-        // Step 2: Initialize and register ExecuteBuyPresenter (OutputBoundary)
-        ExecuteBuyPresenter executeBuyPresenter = new ExecuteBuyPresenter();
-        ServiceManager.registerService(ExecuteBuyPresenter.class, executeBuyPresenter);
-
-        // Step 3: Initialize ExecuteBuyInteractor with DAO and Presenter
-        ExecuteBuyInputBoundary executeBuyInteractor = new ExecuteBuyInteractor(userDataAccessObject, executeBuyPresenter);
+    public AppBuilder addActionPanels() {
+        addPanel("TradeSimulationPanel", new TradeSimulationPanel());
+        addPanel("TransactionHistoryPanel", new TransactionHistoryPanel());
+        return this;
+    }
 
-        // Step 4: Initialize ExecuteBuyController with ExecuteBuyInteractor and register
-        ExecuteBuyController executeBuyController = new ExecuteBuyController(executeBuyInteractor);
-        ServiceManager.registerService(ExecuteBuyController.class, executeBuyController);
+    /**
+     * Adds the dialog component
+     */
+    public AppBuilder addDialogComponent() {
+        ServiceManager.Instance().registerService(DialogComponent.class, new DialogComponent());
+        return this;
+    }
 
-        // Step 5: Initialize and register ViewHistoryPresenter (OutputBoundary)
-        ViewHistoryPresenter viewHistoryPresenter = new ViewHistoryPresenter();
-        ServiceManager.registerService(ViewHistoryPresenter.class, viewHistoryPresenter);
+    /**
+     * Helper method to add a panel to both the card layout and panels map
+     */
+    private void addPanel(String name, JPanel panel) {
+        panels.put(name, panel);
+        cardPanel.add(panel, name);
+    }
 
-        // Step 6: Initialize ViewHistoryInteractor with DAO and Presenter
-        ViewHistoryInputBoundary viewHistoryInteractor = new ViewHistoryInteractor(userDataAccessObject, viewHistoryPresenter);
+    /**
+     * Initializes all required services
+     */
+    private void initializeServices() {
+        ServiceManager serviceManager = ServiceManager.Instance();
+
+        // 1. Initialize DAOs first
+        InMemoryStockDataAccessObject stockDAO = new InMemoryStockDataAccessObject();
+        InMemoryUserDataAccessObject userDAO = new InMemoryUserDataAccessObject();
+
+        // Register concrete DAOs and their interfaces
+        serviceManager.registerService(StockDataAccessInterface.class, stockDAO);
+        StockMarket.Instance().initialize(stockDAO);
+
+        serviceManager.registerService(InMemoryUserDataAccessObject.class, userDAO);
```

Comment:
The key for `ServiceManager` should only include interface type, as `ServiceManager` is supposed to indicate which interface has which instantiation.

Reply:
Does controller has interface? Interactor and presenter now have interfaces. I'm not sure about how to optimize this manager.